File: generated-code/java/codecraft/TcpReadWrite/src/main/java/trans_gen_test/model/AttackProperties.java
package trans_gen_test.model;

import trans_gen_test.util.StreamUtil;

/**
 * Entity's attack properties
 */
public class AttackProperties {
    /**
     * Maximum attack range
     */
    private int attackRange;

    /**
     * Maximum attack range
     */
    public int getAttackRange() {
        return attackRange;
    }

    /**
     * Maximum attack range
     */
    public void setAttackRange(int value) {
        this.attackRange = value;
    }
    /**
     * Damage dealt in one tick
     */
    private int damage;

    /**
     * Damage dealt in one tick
     */
    public int getDamage() {
        return damage;
    }

    /**
     * Damage dealt in one tick
     */
    public void setDamage(int value) {
        this.damage = value;
    }
    /**
     * If true, dealing damage will collect resource from target
     */
    private boolean collectResource;

    /**
     * If true, dealing damage will collect resource from target
     */
    public boolean isCollectResource() {
        return collectResource;
    }

    /**
     * If true, dealing damage will collect resource from target
     */
    public void setCollectResource(boolean value) {
        this.collectResource = value;
    }

    public AttackProperties(int attackRange, int damage, boolean collectResource) {
        this.attackRange = attackRange;
        this.damage = damage;
        this.collectResource = collectResource;
    }

    /**
     * Read AttackProperties from input stream
     */
    public static AttackProperties readFrom(java.io.InputStream stream) throws java.io.IOException {
        int attackRange;
        attackRange = StreamUtil.readInt(stream);
        int damage;
        damage = StreamUtil.readInt(stream);
        boolean collectResource;
        collectResource = StreamUtil.readBoolean(stream);
        return new AttackProperties(attackRange, damage, collectResource);
    }

    /**
     * Write AttackProperties to output stream
     */
    public void writeTo(java.io.OutputStream stream) throws java.io.IOException {
        StreamUtil.writeInt(stream, attackRange);
        StreamUtil.writeInt(stream, damage);
        StreamUtil.writeBoolean(stream, collectResource);
    }

    /**
     * Get string representation of AttackProperties
     */
    @Override
    public String toString() {
        StringBuilder stringBuilder = new StringBuilder("AttackProperties { ");
        stringBuilder.append("attackRange: ");
        stringBuilder.append(String.valueOf(attackRange));
        stringBuilder.append(", ");
        stringBuilder.append("damage: ");
        stringBuilder.append(String.valueOf(damage));
        stringBuilder.append(", ");
        stringBuilder.append("collectResource: ");
        stringBuilder.append(String.valueOf(collectResource));
        stringBuilder.append(" }");
        return stringBuilder.toString();
    }
}

File: generated-code/kotlin/codecraft/FileReadWrite/src/main/kotlin/trans_gen_test/model/AttackProperties.kt
package trans_gen_test.model

import trans_gen_test.util.StreamUtil

/**
 * Entity's attack properties
 */
class AttackProperties {
    /**
     * Maximum attack range
     */
    var attackRange: Int
    /**
     * Damage dealt in one tick
     */
    var damage: Int
    /**
     * If true, dealing damage will collect resource from target
     */
    var collectResource: Boolean

    constructor(attackRange: Int, damage: Int, collectResource: Boolean) {
        this.attackRange = attackRange
        this.damage = damage
        this.collectResource = collectResource
    }

    /**
     * Write AttackProperties to output stream
     */
    @Throws(java.io.IOException::class)
    fun writeTo(stream: java.io.OutputStream) {
        StreamUtil.writeInt(stream, attackRange)
        StreamUtil.writeInt(stream, damage)
        StreamUtil.writeBoolean(stream, collectResource)
    }

    /**
     * Get string representation of AttackProperties
     */
    override fun toString(): String {
        var stringBuilder = StringBuilder("AttackProperties { ")
        stringBuilder.append("attackRange: ")
        stringBuilder.append(attackRange)
        stringBuilder.append(", ")
        stringBuilder.append("damage: ")
        stringBuilder.append(damage)
        stringBuilder.append(", ")
        stringBuilder.append("collectResource: ")
        stringBuilder.append(collectResource)
        stringBuilder.append(" }")
        return stringBuilder.toString()
    }

    companion object {
        /**
         * Read AttackProperties from input stream
         */
        @Throws(java.io.IOException::class)
        fun readFrom(stream: java.io.InputStream): AttackProperties {
            var attackRange: Int
            attackRange = StreamUtil.readInt(stream)
            var damage: Int
            damage = StreamUtil.readInt(stream)
            var collectResource: Boolean
            collectResource = StreamUtil.readBoolean(stream)
            return AttackProperties(attackRange, damage, collectResource)
        }
    }
}